tests/test_hassos.py

```diff
@@ -0,0 +1,22 @@
+"""Test Home Assistant OS functionality."""
+
+from awesomeversion import AwesomeVersion
+import pytest
+
+from supervisor.coresys import CoreSys
+
+# pylint: disable=protected-access
+
+
+@pytest.mark.asyncio
+async def test_ota_url_generic_x86_64_rename(coresys: CoreSys) -> None:
+    """Test download URL generated."""
+
+    coresys.hassos._board = "intel-nuc"
+    coresys.hassos._version = AwesomeVersion("5.13")
+    await coresys.updater.fetch_data()
+
+    version6 = AwesomeVersion("6.0")
+    url = coresys.updater.ota_url.format(version=str(version6), board="generic-x86-64")
```

Comment:
There is no need to wrap it in a str here
```
>>> OTA_URL = "https://github.com/home-assistant/operating-system/releases/download/{version}/hassos_{board}-{version}.raucb"
>>> version6 = AwesomeVersion("6.0")
>>> OTA_URL.format(version=version6, board="generic-x86-64")
'https://github.com/home-assistant/operating-system/releases/download/6.0/hassos_generic-x86-64-6.0.raucb'
```
```suggestion
    url = coresys.updater.ota_url.format(version=version6, board="generic-x86-64")
```

Reply:
This comes from hassos.py: https://github.com/home-assistant/supervisor/blob/main/supervisor/hassos.py#L69

Does that work in every and all cases? E.g. how does Python knows that format requires a string there? Could it be that if AwesomeVersion implements `__int__`, this is picked over `__str__`?

Reply:
By experimentation it seems that `__str__` really is the preferred method, over `__int__` and `__repr__`.

Reply:
It does not work everywhere, it works here because of <https://github.com/ludeeus/awesomeversion/blob/main/awesomeversion/awesomeversion.py#L83-L84>, if it had `__init__` (which it can't), `__str__` would still be used